pyartcd/pyartcd/pipelines/gen_assembly.py

```diff
@@ -164,50 +156,75 @@ async def _gen_assembly_from_releases(self, candidate_nightlies: Iterable[str])
+            self._logger.warning(
+                "[DRY RUN] Would have created pull-request with head '%s', base '%s' title '%s', body '%s'",
+                head, base, title, body)
+
+            if self.auto_trigger_build_sync:
+                self._logger.info(
+                    "[DRY RUN] Would have triggered build-sync with the PR assembly definition")
             d = {"html_url": "https://github.example.com/foo/bar/pull/1234", "number": 1234}
             result = namedtuple('pull_request', d.keys())(*d.values())
             return result
+
+        # Clone ocp-build-data
+        build_data_path = self._working_dir / "ocp-build-data-push"
+        build_data = GitRepository(build_data_path, dry_run=self.runtime.dry_run)
+        await build_data.setup(ocp_build_data_repo_push_url)
+        await build_data.fetch_switch_branch(branch, self.group)
+
+        # Make changes to releases.yml and push changes to auto-gen-assembly branch
+        releases_yaml_path = build_data_path / "releases.yml"
+        releases_yaml = yaml.load(releases_yaml_path) if releases_yaml_path.exists() else {}
+        releases_yaml = merge_objects(assembly_definition, releases_yaml)
+        yaml.dump(releases_yaml, releases_yaml_path)
         pushed = await build_data.commit_push(f"{title}\n{body}")
-        result = None
-        if pushed:
-            owner = "openshift-eng"
-            repo = "ocp-build-data"
-            api = GhApi(owner=owner, repo=repo, token=self._github_token)
-            existing_prs = api.pulls.list(state="open", base=base, head=head)
-            if not existing_prs.items:
-                result = api.pulls.create(head=head, base=base, title=title, body=body, maintainer_can_modify=True)
-            else:
-                pull_number = existing_prs.items[0].number
-                result = api.pulls.update(pull_number=pull_number, title=title, body=body)
-        else:
+        if not pushed:
             self._logger.warning("PR is not created: Nothing to commit.")
+            return None
+
+        # Create a pull request
+        _, owner, repo = split_git_url(ocp_build_data_repo_push_url)
+        api = GhApi(owner=owner, repo=repo, token=self._github_token)
+        existing_prs = api.pulls.list(state="open", base=base, head=head)
+        if not existing_prs.items:
+            result = api.pulls.create(head=head, base=base, title=title, body=body, maintainer_can_modify=True)
+        else:
+            pull_number = existing_prs.items[0].number
+            result = api.pulls.update(pull_number=pull_number, title=title, body=body)
+
+        # Trigger build-sync
+        if self.auto_trigger_build_sync:
+            self._logger.info("Triggering build-sync")
+            build_version = self.group.split("-")[1]  # eg: 4.14 from openshift-4.14
+            # we're not passing doozer_data_path to build-sync because we always create branch on the base repo
+            start_build_sync(build_version=build_version,
```

Comment:
`result` has the pr number, we can check the build_sync job result, add the comment here also notify in the same thread.

Reply:
`build-sync` is started in non-blocking mode. I think this is fine, as we don't want to have these two jobs tightly coupled. This means gen-assembly won't care about `build-sync` result, it will just fire it and exit

Reply:
I think it should care about the result, like if success just comments with `api.issues.create_comment(issue_number=pr_number, body=text_body)` after

Reply:
slack messages can also move to here, so it can append in the thread